service/src/main/scala/za/co/absa/logingw/rest/config/ConfigValidatable.scala

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright 2023 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.logingw.rest.config
+
+trait ConfigValidatable {
+
+  /** May throw ConfigValidationException */
+  def validate(): Unit
```

Comment:
Have you thought about using `Try[Unit]` at this level? Or a custom trait `ConfigValidationResult` and case object `ConfigValidationSuccess` and case class `ConfigValidationError(reason)`

Reply:
Thanks for the idea, I followed it.
Originally, I just wanted the config validation to fail on an invalid config and that was it. It still does, but now there is a wrapper (_at the end of the world_ sort of), but the validation is done more idiomatically.

Reply:
Looks cool!